src/commonMain/kotlin/exchange.dydx.abacus/processor/wallet/account/SubaccountProcessor.kt

```diff
@@ -238,15 +277,31 @@ internal open class SubaccountProcessor(parser: ParserProtocol) : BaseProcessor(
         return Pair(existing, false)
     }
 
-    private fun receivedFills(
+    private fun processFills(
+        subaccount: InternalSubaccountState,
+        payload: List<IndexerFillResponseObject>?,
+        reset: Boolean,
+    ): InternalSubaccountState {
+        val newFills = fillsProcessor.process(
+            existing = if (reset) null else subaccount.fills,
+            payload = payload ?: emptyList(),
+            subaccountNumber = subaccount.subaccountNumber,
+        )
+        if (subaccount.fills != newFills) {
```

Comment:
do we need this check here? why not just always set?

Reply:
It's the same pattern we had before where we don't update the object unless there is a diff.  Downstream processing relies on this when constructing the change set for web FE.